src/rev_ai/streamingclient.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+"""StreamingClient tool used for streaming services"""
+import websocket
+import threading
+import six
+import json
+
+
+def on_error(error):
+    raise error
+
+
+def on_close(code, reason):
+    print("Connection Closed. Code : {}; Reason : {}".format(code, reason))
+
+
+def on_connected(job_id):
+    print('Connected, Job ID : {}'.format(job_id))
+
+
+class RevAiStreamingClient():
+    def __init__(self,
+                 access_token,
+                 config,
+                 version='v1alpha',
+                 on_error=on_error,
+                 on_close=on_close,
+                 on_connected=on_connected):
+        """Constructor for Streaming Client
+
+        :param access_token: access token which authorizes all requests and
+            links them to your account. Generated on the settings page of your
+            account dashboard on Rev.ai.
+        :param config: a MediaConfig object containing audio information.
+            See MediaConfig.py for more information
+        :param version (optional): version of the streaming api to be used
+        :param on_error (optional): function to be called when recieving an
+            error from the server
+        :param on_close (optional): function to be called when the websocket
+            closes
+        :param on_connected (optional): function to be called when the websocket
+            and thread starts successfully
+        """
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        if not config:
+            raise ValueError('config must be provided')
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = 'wss://api.rev.ai/speechtotext/{}/stream'. \
+            format(version)
+        self.on_error = on_error
+        self.on_close = on_close
+        self.on_connected = on_connected
+        self.client = websocket.WebSocket(enable_multithread=True)
+
+    def start(self, generator):
+        """Function to connect thde websocket to the URL and start the response
+            thread
+
+        :param generator: generator object that yields binary audio data
+        """
+        url = self.base_url + '?access_token={}'.format(self.access_token) \
+            + '&content_type={}'.format(self.config.get_content_type_string())
+        try:
+            self.client.connect(url)
+            self._start_send_data_thread(generator)
+            return self._get_response_generator()
+        except Exception as e:
```

Comment:
oh right because you can override the handler, derp

Reply:
derp, it's because you want to be able to override the behavior

Reply:
This provides the user the ability to have a custom error function depending on the error raised by the websocket connection. The default is set to raise. I have it as a generic exception because I'm sure there are multiple possible errors to be obtained via initializing the websocket connection.